File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> cal(int num, int n) {
    if(num == 0) {
        return vector<int>(1, 0);
    }
    
    vector<int> tmp;
    while(num > 0) {
        tmp.push_back(num % n);
        num /= n;
    }
    
    reverse(tmp.begin(), tmp.end());
    
    return tmp;
}

string solution(int n, int t, int m, int p) {
    string answer = "";
    
    vector<int> v;
    int num = 0;
    while(v.size() <= t * m) {
        vector<int> tmp = cal(num, n);
        v.insert(v.end(), tmp.begin(), tmp.end());
        num++;
    }
    
    for(int idx = 0; idx < t; idx++) {
        int cur_num = v[p + idx * m - 1];
        if(cur_num < 10) {
            answer += to_string(cur_num);
        }
        else {
            answer += 'A' + (cur_num % 10);
        }
    }
    
    return answer;
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def convert(num: int, base: int) -> str:
    if num == 0:
        return '0'
    
    ret = ''
    over_nine = ['A', 'B', 'C', 'D', 'E', 'F']
    while num:
        leftover = num % base
        if leftover < 10:
            ret = str(leftover) + ret
        else:
            ret = over_nine[leftover - 10] + ret
            
        num //= base
        
    return ret

def solution(n, t, m, p):
    total_string = ''
    cur_num = 0
    
    while len(total_string) < t * m:
        total_string = total_string + convert(cur_num, n)
        cur_num += 1  
        
    answer = ''
    for _ in range(t):
        answer += total_string[p - 1]
        p += m
        
    return answer